lib/AppInfo/Application.php

```diff
@@ -54,32 +49,6 @@ public function register(IRegistrationContext $context): void {
 	}
 
 	public function boot(IBootContext $context): void {
-		$context->injectFn(Closure::fromCallable([$this, 'registerNavigation']));
 		Util::addStyle(self::APP_ID, 'gitlab-search');
 	}
-
-	public function registerNavigation(IUserSession $userSession): void {
-		$user = $userSession->getUser();
-		if ($user !== null) {
-			$userId = $user->getUID();
-			$container = $this->getContainer();
-
-			if ($this->config->getUserValue($userId, self::APP_ID, 'navigation_enabled', '0') === '1') {
-				$adminOauthUrl = $this->config->getAppValue(Application::APP_ID, 'oauth_instance_url', self::DEFAULT_GITLAB_URL) ?: self::DEFAULT_GITLAB_URL;
-				$gitlabUrl = $this->config->getUserValue($userId, self::APP_ID, 'url', $adminOauthUrl) ?: $adminOauthUrl;
-				$container->get(INavigationManager::class)->add(function () use ($container, $gitlabUrl) {
-					$urlGenerator = $container->get(IURLGenerator::class);
-					$l10n = $container->get(IL10N::class);
-					return [
-						'id' => self::APP_ID,
-						'order' => 10,
-						'href' => $gitlabUrl,
-						'target' => '_blank',
-						'icon' => $urlGenerator->imagePath(self::APP_ID, 'app.svg'),
-						'name' => $l10n->t('GitLab'),
-					];
-				});
```

Comment:
we can have multiple instances here itself without external sites app. Your multi-config refactor can have the name, id, icon for all the instances which would make them unique.

Reply:
No that won't work since we do not have the icon of the instance (which could still be the default gitlab icon) and the name would always be "GitLab". That's entirely useless if you have 5 instances connected. I discussed this with @julien-nc and this was the solution we came up with that still provides decent UX

Reply:
both could be grabbed when connecting to the instance but sure this works too